ledger/ledger-utxo/src/main/kotlin/net/corda/v5/ledger/utxo/StateAndRef.kt

```diff
@@ -0,0 +1,17 @@
+package net.corda.v5.ledger.utxo
+
+import net.corda.v5.base.annotations.CordaSerializable
+import net.corda.v5.base.annotations.DoNotImplement
+
+/**
+ * Represents a composition of a [TransactionState] and a [StateRef].
+ *
+ * @property state The [TransactionState] component of the current [StateAndRef].
+ * @property ref The [StateRef] component of the current [StateAndRef].
+ */
+@DoNotImplement
+@CordaSerializable
+interface StateAndRef<out T : ContractState> {
```

Comment:
We should define a better name and get that issue closed and decided.

Reply:
This is a familiar type for C4 developers so we should leave alone, though needs better doc comments